File: 9.palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0 || (x != 0 && x%10 == 0))
            return false;
        int sum = 0;
        while (x > sum) {
            sum = sum * 10 + x % 10;
            x /= 10;
        }
        return x == sum || x == sum/10;  //不能写成x*10 == sum， 反例：小于10的正整数
    }
};

File: 9.palindrome-number.js
/*************************************************************************
    @ File Name: 9.palindrome-number.js
    @ Author: Rain
    @ Mail: yutiangan@yangqianguan.com
    @ Created Time: 六  7/14 17:34:53 2018
 ************************************************************************/

/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
    const xString = x.toString();
    const xStringReverse = xString.split('').reverse().join('');
    return xString === xStringReverse;
};
